compiler/semtypes.nim

```diff
@@ -1226,10 +1227,11 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
     if hasType:
       typ = semParamType(c, a[^2], constraint)
       # TODO: Disallow typed/untyped in procs in the compiler/stdlib
-      if kind == skProc and (typ.kind == tyTyped or typ.kind == tyUntyped):
+      if kind in {skProc, skFunc} and (typ.kind == tyTyped or typ.kind == tyUntyped):
```

Comment:
```suggestion
      if kind in {skProc, skFunc} and typ.kind in {tyTyped, tyUntyped}:
```

Reply:
I can defer the change of this file because other places have the same problem: missing `skFunc`